src/providers/ldap/ldap_options.c

```diff
@@ -747,9 +747,9 @@ errno_t common_parse_search_base(TALLOC_CTX *mem_ctx,
                 ret = EINVAL;
                 goto done;
             }
-            talloc_zfree(ldn);
 
             /* Set the search base DN */
+            search_bases[i]->ldb_basedn = talloc_steal(search_bases[i], ldn);
```

Comment:
Hi,

you have to cover the case as well where there is only 1 element, i.e. a single DN. I guess the most easy way would be to add a simliar line like the one about to `sdap_create_search_base()`.

bye,
Sumit

Reply:
Oh! I missined this case was treated separately.

I did as you proposed.